### PR TITLE
Add a way to control where something is inserted into the buffer

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/hbbft.erl [hunk 1]
```diff
@@ -879,3 +909,53 @@ decode_list(<<Length:24/integer-unsigned-little, Entry:Length/binary, Tail/binar
...
+buf_insert(BufIn, Tx, Cmp) ->
+    {_, {InsertionPosition, BufOutReversed}} =
+        lists:foldl(
+            fun (Tx_I, {I, {J, BufOut}}) ->
+                case Cmp(Tx_I) of
+                    true ->
+                        {I + 1, {J, [Tx_I | BufOut]}};
+                    false ->
+                        %% TODO Insert BEFORE or AFTER the point?
+                        {I + 1, {I + 1, [Tx, Tx_I | BufOut]}}
```
Comment (xandkar) [inline src/hbbft.erl:925]: This is wrong - has to terminate after insertion.

**Review by andymck — APPROVED** [review]
lgtm

File: src/hbbft.erl [hunk 2]
```diff
@@ -219,14 +220,43 @@ start_on_demand(Data) ->
     {Data, already_started}.
 
 %% someone submitting a transaction to the replica set
--spec input(hbbft_data(), binary()) -> {hbbft_data(), ok | {send, [rbc_wrapped_output()]} | full}.
-input(Data = #hbbft_data{buf = Buf, max_buf = MaxBuf}, Txn) when
-    is_binary(Txn), length(Buf) < MaxBuf
+-spec input(hbbft_data(), binary()) ->
```
Comment (evanmcc) [inline src/hbbft.erl:223]: fairly certain that it's safe to omit this spec for the defaulting arity

**Review by evanmcc — APPROVED** [review]
could move to types or remove the first spec, either way this looks OK to go in.

**Pull request merged**